### PR TITLE
add formatter workflow

### PR DESCRIPTION
**Pull request by ajwheeler**

### CODE REVIEW

**Comment by ajwheeler** [comment]
I've opened a issue which addresses the place where this is failing: https://github.com/domluna/JuliaFormatter.jl/issues/860

File: src/linelist.jl [hunk 1]
```diff
@@ -357,9 +357,13 @@ function parse_vald_linelist(f, isotopic_abundances)
...
     Δlog_gf = if scale_isotopes
         refs = if !shortformat #the references are on different lines
+            # this line breaks the code formatter. 
+            # https://github.com/domluna/JuliaFormatter.jl/issues/860
+            #! format: off
             lines[firstline+3 .+ ((0:length(body)-1) .* 4)]
+            #! format: on
         else #references are in the last column
-            body.reference
+              body.reference
```
[the remark this answers is not in the document]
Comment (ajwheeler) [reply src/linelist.jl:366]: It works!

File: src/atmosphere.jl
[the remark this answers is not in the document]
Comment (ajwheeler) [reply src/atmosphere.jl:1]: I think this is not a suggestion just because the diff is too big?

**Comment by ajwheeler** [comment]
Attention contributors to this repository and people with Opinions: now that I have the github actions aspect of this working how I would like, I'm going to format the whole repo with JuliaFormatter and require that all changes maintain the style going forward. (This tooling will automatically suggest the changes, and it's easy to format locally as well.)

This is your chance to tell me that my style choices (`.JuliaFormatter.toml`) are wrong.  Documentation [here](https://domluna.github.io/JuliaFormatter.jl/stable/#Formatting-Options)

**Comment by mabruzzo** [comment]
> edit: really tempted by https://domluna.github.io/JuliaFormatter.jl/stable/#for_in_replacement

As someone who doesn't contribute much new code, my opinion definitely shouldn't matter here. But, my initial gut instinct is that I prefer `for a in 1:10` or `for a = 1:10` over `for a ∈ 1:10` (I think it's that I don't love the idea of using `∈` as builtin language syntax). But again, my opinion shouldn't count here. Plus the beauty of the formatter is that I could write my code one way and the formatter will change it to the standard style.

**Comment by ajwheeler** [comment]
I value your opinion!  I also changed my mind when I realized that it would
be annoying to search for "for x ∈".

**Pull request merged**